app/common/_temp_hooks/useAmountLogic.ts

```diff
@@ -2,18 +2,21 @@ import { useEffect, useState } from 'react';
 
 import { BN, BN_ZERO } from '@polkadot/util';
 
-import { type ITransfer } from '@/shared/api/types';
+import { type IBalance, type ITransfer } from '@/shared/api/types';
 import { toPreciseBalance } from '@/shared/helpers';
 import { type Asset, type Balance } from '@/types/substrate';
 
 type AmountLogicParams = {
-  service: ITransfer;
+  services: {
+    transferService: ITransfer;
+    balanceService: IBalance;
+  };
   asset: Asset;
   balance?: Balance;
   isGift: boolean;
 };
 
-export const useAmountLogic = ({ service, asset, balance, isGift }: AmountLogicParams) => {
+export const useAmountLogic = ({ services, asset, balance, isGift }: AmountLogicParams) => {
```

Comment:
An idea for the future. If we add new types of "amount logic" in the future it might be beneficial to have the following setup to make `useAmountLogic` reusable (we have the similar logic in Nova)

```
interface MaxAmountProvider {
  // amount shown to the user might differ from what we actually use, e.g. when we dont deduct fees from the displayed amount
  funciton forDisplay(): Observable<balance>

function forLogic(): Observable<balance>
}
```
This can be adjusted  based on telenova specifics

Then we can have a decorator-based set of implementations, e.g.

```
class AvailableBalalnceMaxProvider(asset: Observable<Asset>) implements MaxAmountProvider {
// use balance of an asset
}

class FeeDeductionMaxProvider(delegate: MaxAmountProvider, fees: Observable<Fees>) {
// watch for fees and reduce balance received from delegate
}
```

Then have a set of factory functions to be able to compose providers like the following:

```
let provider = maxAvailableFrom(asset).deductFees(fees)
```

This provides highly configurable foundation for amount logic that can be used in almost every situation. E.g. gifts might have a separate implementation of `MaxAvailableProvider` to provide logic related to fees